include/aspect/geometry_model/interface.h

```diff
@@ -144,6 +144,13 @@ namespace aspect
         virtual
         double height_above_reference_surface(const Point<dim> &position) const = 0;
 
+        /**
+         * Converts a Cartesian Point into another coordinate system and returns it
+         * as a Point.
+         */
+        Point<dim>
+        cartesian_to_other_coordinates(const Point<dim> &position,
+                                       const Utilities::Coordinates::CoordinateSystem &coordinate_system) const;
```

Comment:
I agree with @MFraters that this function should not return a `Point<dim>`, but an `std::array<double,dim>` or an instance of `NaturalCoordinates`.